src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -352,6 +352,7 @@ public sealed partial class ReadOnlyMemoryContent : System.Net.Http.HttpContent
     public sealed partial class SocketsHttpHandler : System.Net.Http.HttpMessageHandler
     {
         public SocketsHttpHandler() { }
+        [System.Runtime.Versioning.UnsupportedOSPlatformGuardAttribute("browser")]
         public static bool IsSupported { get { throw null; } }
```

Comment:
Added the annotation, but it seems not used anywhere, within or outside of the project

Reply:
> but it seems not used anywhere

It is a public property on a public class. So customers can call it.

> Added the annotation

How does this work w.r.t. having `[UnsupportedOSPlatform("browser")]` on `SocketsHttpHandler` itself? Won't callers get a warning by referencing the `SocketsHttpHandler.IsSupported` property because the whole `SocketsHttpHandler` class is marked as `UnsupportedOSPlatform("browser")`?

Reply:
Tracked as #49755

Reply:
> It is a public property on a public class. So customers can call it.

Yes, i meant no reference needs an update within runtime

> Won't callers get a warning by referencing the `SocketsHttpHandler.IsSupported` property because the whole `SocketsHttpHandler` class is marked as `UnsupportedOSPlatform("browser")`?

Normally yes, but after adding `UnsupportedOSPlatformGuard("browser")` it will not warn for the property usage anymore, plus the property can be used as a guard